napari_tiff/napari_tiff_metadata.py

```diff
@@ -0,0 +1,395 @@
+                    except Exception:
+                        pass
+                metadata_dict[name] = metadata_value
+    return metadata_dict
+
+
+def get_tiff_metadata(tif: TiffFile) -> dict[str, Any]:
+    """Return napari metadata from largest image series in TIFF file."""
+    # TODO: fix (u)int32/64
+    # TODO: handle complex
+    series = tif.series[0]
+    for s in tif.series:
+        if s.size > series.size:
+            series = s
+    dtype = series.dtype
+    axes = series.axes
+    shape = series.shape
+    page = next(p for p in series.pages if p is not None)
+    extrasamples = page.extrasamples
+
+    rgb = page.photometric in (2, 6) and shape[-1] in (3, 4)
+    name = None
+    scale = None
+    colormap = None
+    contrast_limits = None
+    blending = None
+    channel_axis = None
+    visible = True
+
+    if page.photometric == 5:
+        # CMYK
+        channel_axis = axes.find("S")
+        if channel_axis >= 0 and shape[channel_axis] >= 4:
+            colormap = cmyk_colormaps()
+            name = ["Cyan", "Magenta", "Yellow", "Black"]
+            visible = [False, False, False, True]
+            blending = ["additive", "additive", "additive", "additive"]
+            # TODO: use subtractive blending
+        else:
+            channel_axis = None
+    elif page.photometric in (2, 6) and (
+        page.planarconfig == 2
+        or (page.bitspersample > 8 and dtype.kind in "iu")
+        or (extrasamples and len(extrasamples) > 1)
+    ):
+        # RGB >8-bit or planar, or with multiple extrasamples
+        channel_axis = axes.find("S")
+        if channel_axis >= 0 and shape[channel_axis] > 2:
+            rgb = False
+            visible = [True, True, True]
+            colormap = ["red", "green", "blue"]  # rgb_colormaps()
+            name = ["Red", "Green", "Blue"]
+            blending = ["additive", "additive", "additive"]
+        else:
+            channel_axis = None
+    elif (
+        page.photometric in (0, 1)
+        and extrasamples
+        and any(sample > 0 for sample in extrasamples)
+    ):
+        # Grayscale with alpha channel
+        channel_axis = axes.find("S")
+        if channel_axis >= 0:
+            visible = [True]
+            colormap = ["gray"]
+            name = ["Minisblack" if page.photometric == 1 else "Miniswhite"]
+            blending = ["additive"]
+        else:
+            channel_axis = None
+
+    if channel_axis is not None and extrasamples:
+        # add extrasamples
+        for sample in extrasamples:
+            if sample == 0:
+                # UNSPECIFIED
+                visible.append(False)  # hide by default
+                colormap.append("gray")
+                name.append("Extrasample")
+                blending.append("additive")
+            else:
+                # alpha channel
+                # TODO: handle ASSOCALPHA and UNASSALPHA
+                visible.append(True)
+                colormap.append(alpha_colormap())
```

Comment:
I do not see any comment explaining why we would like to provide transparent color map.